File: Backend/ProjectsService/ProjectsService.Domain/Entities/ProjectReport.cs
using ProjectsService.Domain.Enums;
using ProjectsService.Domain.Primitives;

namespace ProjectsService.Domain.Entities;

public class ProjectReport : Entity
{
    public string? Description { get; set; }
    public ReportStatus Status { get; set; }
    public Guid ProjectId { get; set; }
    public Project? Project { get; set; }
    public Guid ReporterUserId { get; set; }
    public Guid? ReviewerUserId { get; set; }
}

File: Frontend/src/app/core/interfaces/project/project-report.ts
import { Project } from "./project.interface";
import { EmployerUser } from "../employer/employer-user.interface";

export interface ProjectReport {
  id: string;
  description: string | null;
  status: ReportStatus;
  projectId: string;
  project: Project | null;
  reporterUserId: string;
  reviewerUserId: string | null;
}

export enum ReportStatus {
  Sent = 0,
  Reviewed = 1,
  Rejected = 2,
}